apps/angular/projection/src/app/ui/card/card.component.ts

```diff
@@ -1,34 +1,33 @@
-import { NgFor, NgIf } from '@angular/common';
-import { Component, Input } from '@angular/core';
-import { randStudent, randTeacher } from '../../data-access/fake-http.service';
-import { StudentStore } from '../../data-access/student.store';
-import { TeacherStore } from '../../data-access/teacher.store';
-import { CardType } from '../../model/card.model';
+import { NgFor, NgIf, NgTemplateOutlet } from '@angular/common';
+import {
+  ChangeDetectionStrategy,
+  Component,
+  ContentChild,
+  Input,
+  TemplateRef,
+} from '@angular/core';
 import { ListItemComponent } from '../list-item/list-item.component';
+import { Teacher } from '../../model/teacher.model';
+import { Student } from '../../model/student.model';
+import { City } from '../../model/city.model';
+
+type item = Teacher | Student | City;
 
 @Component({
   selector: 'app-card',
   templateUrl: './card.component.html',
   standalone: true,
-  imports: [NgIf, NgFor, ListItemComponent],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  imports: [NgIf, NgFor, ListItemComponent, NgTemplateOutlet],
 })
 export class CardComponent {
-  @Input() list: any[] | null = null;
-  @Input() type!: CardType;
+  @Input() list: item[] | null = null;
   @Input() customClass = '';
+  @ContentChild('listItem', { read: TemplateRef }) listItem: TemplateRef<{
```

Comment:
To make it more safe, prefer using directive over magic string

Reply:
Im currently on challenge 4 , its intense , im sweating , i  have read ur blogs its awesome